### ISSUE TITLE
Automatically throw BadRequestException on failed parameter parsing as well as javax.validation.constraints violations

### ISSUE THREAD

**Issue by agargenta** [jetrs/jetrs#41]
[Jakarta RESTful Web Services](https://jakarta.ee/specifications/restful-ws/3.0/jakarta-restful-ws-spec-3.0#validation) states that “This version of JAX-RS introduces support for declarative validation based on the Bean Validation specification”. [Jakarta Bean Validation specification](https://jakarta.ee/specifications/bean-validation/2.0/bean-validation_2.0) in turn declares a number of [javax.validation.constraints (Java(TM) EE 7 Specification APIs)](https://docs.oracle.com/javaee%2F7%2Fapi%2F%2F/javax/validation/constraints/package-summary.html) , which do not appear to be supported by JetRS.

For example, given:
```java
@GET
@Path("foos")
...
public @api.Foo List<api.Foo> getFoos
   ..., 
   @Nullable @Positive @Digits(integer=4, fraction = 0) @QueryParam("limit") final Integer limit
  ) ... {
  ...
}
```
JetRS will return `limit` as `null` for invalid integers (e.g. `.../foos?limit=bar`) as return `limit` as-is even for negative values (e.g. `.../foos?limit=-10`). Note, I tried other constraints as well (e.g. `@Min`), and I could not get them to work.

My expectation would be for JetRS to automatically throw `BadRequestException` when any of the consonants are not met, as well as when `<Type>.valueOf("string")` raises an exception (in this case `Integer.valueOf("bar")` throws `NumberFormatException`, which should be wrapped as `BadRequestException`).

**Comment by safris** [comment]
@agargenta, JetRS supports JAX-RS v2.0 (not v3), which is stated [here](https://github.com/jetrs/jetrs?tab=readme-ov-file#introduction).

> My expectation would be for JetRS to automatically throw `BadRequestException`.

Does the specification explicitly state that a `BadRequestException` must be thrown in the situations you outline?

**Comment by agargenta** [comment]
Good point. [7.6. Validation and Error Reporting](https://jakarta.ee/specifications/restful-ws/3.0/jakarta-restful-ws-spec-3.0#validation_and_error_reporting) states that
* `jakarta.validation.ValidationException` should be mapped to HTTP 500
* `jakarta.validation.ConstraintViolationException` should be mapped to HTTP 500
* Otherwise, it should be mapped to HTTP 400

Would that work?

**Comment by safris** [comment]
@agargenta, I can work on this when my own need for this functionality arises. I am also open to review PRs from you or your team.

**Comment by agargenta** [comment]
Thanks @safris for being open to my request in principle. This is not a significant-enough pain for us at the moment to warrant opening up a PR to address it, but we'll definitely consider it if/when the situation changes.
